### DISCUSSION TITLE
First test, first observations...

### DISCUSSION THREAD

**Discussion by sspieser** [doudar/SmartSpin2k#138] [General]
Hello,
I have done the wiring (I guess), and got the last source code from git tonight.
So I tried Zwift - Emily's short mix training plan. Some observations:

- Power : the "Double powermeter output" does not work
- Once Zwift starts the first interval, the stepper slowly tightened the know to the max, then finally gets back to the opposite direction to the minimum possible... and that's it

Thanks for any advice ;-)

**Comment by doudar** [comment]
Congratulations on getting it together!  Any pictures?

How's normal incline mode work (non -erg?)

If your power meter isn't working correctly due to the double power option, the issue above is probably caused by the "ERG black hole".

We've made huge changes, I'm guessing you compiled from master that had those recently incorporated. I'll check into it. 

If you'd like, you could try this old binary where the double power should work:
https://github.com/doudar/OTAUpdates/blob/main/firmware.bin

I'll look into the double power setting.

**Reply by doudar** [reply]
Yes, I just verified that we missed inserting double power into the new calculations. Master should be updated soon.

**Comment by doudar** [comment]
I just added that back in and created a PR to master for the change. Thanks for spotting it @sspieser !

Hopefully that fixes your ERG problems.

**Reply by sspieser** [reply]
Thx a lot : I just compiled the new master... I'll try as soon as I have some time again! That's my first wiring, it's awfull and do not fit into the box ; I will try to do it better and take pictures :-)

**Comment by jkbaker4** [comment]
Its Alive - :)

A couple of things that I need to work on
- Speed in sufferfest is faster than I am going - Watts and Cadence show correct (82 & 80) but the speed is showing 30mph should be around 20
- I do not know if the auto resistance is working yet.  Is there anyway to test?
- I need something to get rid of the 2x4.  Wish that I was better at 3d - measurement between spin2k and vertical bar is
    -  top distance - 115mm, 
    - bottom distance 105mm.  
    - Vertical is 66mm wide and tapers from 20mm front, 30mm at back

![Spin2k Top](https://user-images.githubusercontent.com/40582757/113482453-d83a2480-946c-11eb-84e7-2ebac8b203c6.jpg)
![Spin2k Side](https://user-images.githubusercontent.com/40582757/113482458-da9c7e80-946c-11eb-952d-c48425bef46e.jpg)

**Reply by fireyeti** [reply]
The second generation "[strap](https://github.com/doudar/SmartSpin2k/blob/develop/Hardware/Mounting_StrapV2.STL)" will reach the head tube of your spin bike and engage with the forward slot on the SmartSpin. It is about 18 cm in length. The SmartSpin does not require a secure fit or strength with the frame of the spin bike, it just needs a fixed point to prevent rotation of the main control box. A gentle fit allows the SmartSpin to be adjusted if it is not exactly centered on the knob as well.

The print and build look great! What type of 3d printer are you using @jkbaker4 ?

**Reply by jkbaker4** [reply]
I printed the 2nd strap and its not long enough.  The distance between the head tube and the forward slot is 180mm.  I would need a strap that is at least 190mm (200mm would be best).  Also should the strap be printed in PLA or something else?

**Reply by fireyeti** [reply]
Looks like @doudar has you covered. Sorry, I misread your measurements as the distance to the front of the head tube, but I see that was my mistake!

**Comment by jkbaker4** [comment]
Ok thanks I will try the other strap.  As for my printer it’s a stock Ender 3 Pro.  I am amazed at how well it works.  Almost done with my first spool of PLA after a month :)

**Reply by doudar** [reply]
I'll design a longer one and get it uploaded today.

PETG or nylon is probably best for those as they need to flex without breaking. Depending on what type of pla it might not last very long .

**Reply by doudar** [reply]
Here's a longer strap for you. Legs are 180mm, overall length should be around 205. Let me know how it works!

Also, for printing these, I like to make a brim on the inside (so it sticks to the plate better), and I use 145c with an 85c build plate on my modified CR-10 with Overture PETG. 

[Mounting_StrapV2_XL.zip](https://github.com/doudar/SmartSpin2k/files/6258344/Mounting_StrapV2_XL.zip)

**Reply by jkbaker4** [reply]
Ok, thanks I will print it this afternoon.

**Reply by doudar** [reply]
Were you able to get the strap to work with pla?

**Reply by jkbaker4** [reply]
Yes, I installed it this morning and its working fine.  I have to see what I can do able the angle as the cup and the insert are binding a bit.  I will send photo's later.

**Reply by jkbaker4** [reply]
Attached is a photo of my setup on an Echelon.  Tilting the unit and pushing the strap down, allowed the spinner to be close to the correct position.  I also figured out my speed issue.  If I use the calculated on sufferfest it shows more accurately.  

My final question is related to ERG mode.  In Erg mode the Sufferfest app will automatically control the resistance on your trainer based on the target wattage we want you to hit.  What I do not see happening is any changes.  I can make changes based on hitting the buttons but nothing is happening automatically.

Here is a little information from the log file...Let me know if you need more..

ERG MODE Target: 82 Current: 0 Incline: 14.44

f0 d1 09 00 d1 00 00 00 18 00 36 00 e9 <- 0bf669f1 | 0bf669f4 | ECH:[ CD(54.00) ]
23 00 00 00 00 18 00 fa 6f <-- CPMC sent
00 00 00 00 00 <-- HR sent
f0 d1 09 00 d1 00 00 00 18 00 36 00 e9 <- 0bf669f1 | 0bf669f4 | ECH:[ CD(54.00) ]
23 00 00 00 00 19 00 6b 74 <-- CPMC sent
00 00 00 00 00 <-- HR sent
f0 d1 09 00 d3 00 00 00 1a 00 36 00 ed <- 0bf669f1 | 0bf669f4 | ECH:[ CD(54.00) ]
23 00 00 00 00 1a 00 dc 78 <-- CPMC sent
00 00 00 00 00 <-- HR sent
f0 d1 09 00 d3 00 00 00 1a 00 36 00 ed <- 0bf669f1 | 0bf669f4 | ECH:[ CD(54.00) ]
23 00 00 00 00 1b 00 4d 7d <-- CPMC sent
00 00 00 00 00 <-- HR sent
f0 d1 09 00 d5 00 00 00 1b 00 34 00 ee <- 0bf669f1 | 0bf669f4 | ECH:[ CD(52.00) ]
23 00 00 00 00 1c 00 ea 81 <-- CPMC sent
00 00 00 00 00 <-- HR sent 5 52 0 <-- From APP
ERG MODE Target: 82 Current: 0 Incline: 15.26

![Echelon Setup](https://user-images.githubusercontent.com/40582757/114081760-f5129580-987a-11eb-905f-d3be98e26610.jpg)

**Reply by fireyeti** [reply]
@jkbaker4 the setup is looking good! We are trying to think through the best design to keep the SS2K stable on this design. I have spent a little time at my local Costco checking out the design :-). 

A few options we thought of was to have a single sided strap going through both notches on the side of the SS2K, then resting flush against the head tube. The strap would then be secured to the head tube in some manner, my suggestion was just a sturdy rubber band, but you could design a sleeve which attached to the strap as well if you're handy with .stl files. Here are a couple drawings. 
![camphoto_1297389768](https://user-images.githubusercontent.com/76080977/114102056-a96ce580-9894-11eb-92d4-92aa972e8f69.jpg)
![camphoto_824023566](https://user-images.githubusercontent.com/76080977/114102061-aa9e1280-9894-11eb-82cf-cfe7aadea5d2.jpg)

If you're ready to abandon the strap you have printed, you can straighten the part that wraps around the vertical head tube by adding heat to the area you want to bend in order to have the strap go through two notches on the side of the SS2K. 

Number one thing is to have the SS2k perpendicular to the axis of resistance knob rotation. Good luck!

**Reply by jkbaker4** [reply]
What I was thinking is something more like this...

![SmartSpin Add-On](https://user-images.githubusercontent.com/40582757/114106526-f2289c80-989c-11eb-9c97-5a8442969f7c.JPG)


But I suck at stl files or have not found a good editor.  What do recommend for editing stl files (I am a IT guy (developer)).

**Reply by doudar** [reply]
Honestly, the best is to just use fusion 360 or SolidWorks (there are discounts available that bring it to ~$50/yr) and design the part to export to an .stl

.stl can be modified by a program called mesh mixer but in my experience it’s hard to get the desired results as stl’s are sort of like .pdf’s - not really made for editing.

**Reply by doudar** [reply]
> What I was thinking is something more like this...
> 
> ![SmartSpin Add-On](https://user-images.githubusercontent.com/40582757/114106526-f2289c80-989c-11eb-9c97-5a8442969f7c.JPG)
> 
> But I suck at stl files or have not found a good editor. What do recommend for editing stl files (I am a IT guy (developer)).

So, I was just going to start looking into designing a mount like this but once it's applied to reality, I think it is going to take a ton of material and might not work that well on this bike as the distances appear to be very similar. Check out my notation below. 

![image](https://user-images.githubusercontent.com/17362216/114273126-8e0df180-99de-11eb-88c5-dbdc394f6513.png)

One of the problems you *might* be having is that the SS2K doesn't look perpendicular to the knob. Try pushing it down to "seat" it onto the knob better. If it's too loose or too tight, we may have to tweak that insert design more. It really should be sitting at exactly the angle of the top of the knob, and the knob should provide 80% of the support to keep the ss2k properly positioned. The front strap is only to keep the SS2K from spinning when the knob turns and can extend past the front of the head tube if required to alleviate pressure. 

If you're having an issue with side to side "slop" at the head tube, possibly you could try some foam pipe insulation. If that helps your issues, I can make a narrower radius at the front for the XL strap.

If the strap needs additional stiffness, that's something easily accomplished also. I'll just make the front 3/4 of the XL strap wider or add a T profile to the portion that doesn't go into the mounting tabs on the SS2K. 

Let me know your thoughts! Thanks!

**Reply by doudar** [reply]
I've just uploaded a new style mounting design that might work well for you. Let me know! 

The idea of this strap is for the front to pass by the head tube and then get secured (lashed) to the head tube with velcro, rubber bands, rope, tape, etc.

This might make a more stable mount for installations that have a long head tube to SS2K distance.

Anything for dimensions on it is easily adjustable. The front part might need to be a bit beefier yet but I also wanted to try to keep it thin so it attaches easy. If you need anything changed before you print it, let me know. 

https://github.com/doudar/SmartSpin2k/blob/7ebe5c73ab44a7f71848477b43dc71314ea65ccc/Hardware/Mounting_Strap_Rigid.STL

**Reply by jkbaker4** [reply]
After looking at the Flywheel_Brand_Mounting_Strap I think with changes that would work better.   So what I am thinking is making the strap in 2 pieces.  The length of the top X that you have is 115mm.  The distance from the front of the stem to the back strap holder is 250mm (to big to print).  So if you made a front strap 60mm wide and 130mm long that has strap holders that are similar to the case and has two legs that are 55mm high (like the flywheel strap).  Then two straps that are 170mm in length with dog legs that would go into the first strap, that should be long enough to go through both supports. 

Attached is a top view the I mocked up as the words maybe confusing...

Let me know what you think...
![Echelon Strap - JKB](https://user-images.githubusercontent.com/40582757/114281944-42c00700-9a0f-11eb-905a-b64ffc9d27ce.JPG)
..

**Reply by doudar** [reply]
So basically you're thinking a U shaped piece to hold the bottom (loose ends) of the flywheel style strap captive with each leg of the U 130mm Long and the width between the legs 30mm?

If that's correct, I can look at designing it later today.

**Reply by jkbaker4** [reply]
Yes, that is correct 30mm between the legs on the U shaped piece.  The strap holding the SS2K will slide into the lugs on the side of the U shaped strap...

It would probably be easier to make by eliminating the doglegs and use straight lines for the strap for the SS2K.

**Reply by doudar** [reply]
Okay, so you're still thinking about going around the head tube. Got it.

**Reply by jkbaker4** [reply]
Good Morning, attached is my first try at creating a strap.  Please review, I created it as a stl and glb (not knowing if your able to use either to modify either).  Take a look and let me know if I am missing anything or I am off anywhere...
[Echelon Strap.zip](https://github.com/doudar/SmartSpin2k/files/6303534/Echelon.Strap.zip)

**Reply by doudar** [reply]
It looks pretty good!

It will probably work as-is but here are some observations: 
* I'm not sure you want/need the doglegs on the front but without a test fit and ride, it's hard to tell either way. 
* You may consider adding fillets on the inside of the sharp (90 degree) thin corners to reduce mechanical stress build up in those areas. A small fillet will add a surprising amount of strength/flexibility because it allows the stress to spread out a ways on both edges and not be trapped in the corner. 
* I'd move the secondary attachment legs on the SS2K mount to the bottom to facilitate printing without supports.  
* The strap going around the head tube isn't bound by the thickness requirements of fitting through the SS2K mounting loops, so you could consider adding extra thickness to make it more rigid. 

Good job!

edit: also curious what you used to design it with.

**Reply by jkbaker4** [reply]
Thanks for the suggestions...Attached is an update (hopefully I understood everything).

To make the strap I used - https://www.tinkercad.com/ 

Its free, pretty accurate and easy to use.  I tried FreeCad but could not get FreeCad to allow me to make all of the parts.  What I was looking for was something that I could give you an idea as to what I was looking for and found I could create it myself...

[Echelon Strap - Update.stl.zip](https://github.com/doudar/SmartSpin2k/files/6306109/Echelon.Strap.-.Update.stl.zip)

**Reply by doudar** [reply]
Cool - I've used tinkercad a little bit before. 

So what you added are "chamfers" but it should perform the same function. A fillet would be a constant curvature fill in the corners. I'm only pointing out the nomenclature to help.

I overlooked adding them to the front of the SS2K side of the strap previously but good job on adding them there. You might want them on both sides on that piece. 

The only other thing then I'd suggest is moving that piece down so it's all level with the build plate and then you won't need printing supports. Unless you like the aesthetics of having it centered. And if you do like it, it's your design so by all means, go for it! 

If you print it and it works, let me know and we will get it added to the hardware folder. If you can do a pull request, that's best as I think GitHub gives you a little more credit vs me doing it for you and calling you a co-author.

**Reply by marklogan17** [reply]
Hope you don't mind me hijacking here - I've been following this discussion as I'm planning on building for the Echelon Sport as well.  I ordered my parts before noticing that I could order the PCB so I guess I'm building the non-pcb version.  So my question... the non-pcb bottom case doesn't seem to have the same strap holder mounts as on the PCB one.  Am I missing a version somewhere?  I'm sure I'll have lots of other questions in the future :)

**Reply by doudar** [reply]
> Hope you don't mind me hijacking here - I've been following this discussion as I'm planning on building for the Echelon Sport as well. I ordered my parts before noticing that I could order the PCB so I guess I'm building the non-pcb version. So my question... the non-pcb bottom case doesn't seem to have the same strap holder mounts as on the PCB one. Am I missing a version somewhere? I'm sure I'll have lots of other questions in the future :)

Well would you look at that.....

I'll rebuild the .stl and upload. Thanks

**Reply by marklogan17** [reply]
:) I kept looking and trying to figure out what I was missing.  LOL.

**Reply by doudar** [reply]
> :) I kept looking and trying to figure out what I was missing. LOL.

If you can't wait, Here's an older version from @sspieser 's repo with the loops. It's only really missing the updated vertical vents and some other improvements mostly done for easier printing.
https://github.com/sspieser/SmartSpin2k/blob/master/Hardware/Bottom_Case.stl

**Reply by marklogan17** [reply]
Thanks - but Aliexpress is not fast...  I've got plenty of time :)  Works out because @jkbaker4 will have the support all figured out for me.

**Reply by doudar** [reply]
Sometimes we're fast. It's there now. (in develop branch)

**Reply by marklogan17** [reply]
Looks great.  Thank you!  I'm really looking forward to building this.

**Reply by fireyeti** [reply]
I'm sure you've come across the build guides. but if not, they're in the [wiki](https://github.com/doudar/SmartSpin2k/wiki/Building-How-To). We're trying to make it as easy as possible for new users to build, so if there's anything we can do to clarify or help just let us know in a new discussion and I'll work on streamlining things.

**Reply by marklogan17** [reply]
Yes.  I've noticed huge changes even over the last week or two.  I appreciate the effort you guys are making to clarify how this all goes together.  Many questions I had have already been answered by your recent changes.

**Reply by jkbaker4** [reply]
Good Morning, attached is the Echelon Strap - final design and pictures of the one I printed last night (Not the Final) mounted.  

The changes I made are:
- Moved the legs back 15cm
- Removed the 2nd holder on the side and lengthen the first (it was hard to remove the supports)
- Shortened the strap that goes around the SS2K
- Made the fronts of both straps 4mm (add a little strength) 
- Increased the width of the SS2K strap to 63mm

It was very cool though to create the strap, print it and with a few minor adjustments come up with something that will work..

![IMG_3153](https://user-images.githubusercontent.com/40582757/114710883-464bda80-9cfc-11eb-9a28-ccdf85083f98.jpg)
![IMG_3151](https://user-images.githubusercontent.com/40582757/114710885-46e47100-9cfc-11eb-92ac-1d93f06f7570.jpg)
![IMG_3150](https://user-images.githubusercontent.com/40582757/114710887-46e47100-9cfc-11eb-9531-9c66efb171f8.jpg)

[Echelon Strap - Final.stl.zip](https://github.com/doudar/SmartSpin2k/files/6310770/Echelon.Strap.-.Final.stl.zip)

**Reply by marklogan17** [reply]
@jkbaker4 looks great.  What did you use to connect the buttons to the bike handles?

**Reply by jkbaker4** [reply]
I found a clamp that was the correct diameter on thingiverse.  Its on my other computer, so I will upload it tonight.

**Reply by jkbaker4** [reply]
Attached is the STL file for what I used to create the button mounts...

[35mm_cygolite_bar_mount.zip](https://github.com/doudar/SmartSpin2k/files/6314229/35mm_cygolite_bar_mount.zip)

**Reply by marklogan17** [reply]
Great thank you!  I'm printing out the base and then I'll print out your strap to try.  Then just the long wait for the other parts.

**Reply by marklogan17** [reply]
> I've just uploaded a new style mounting design that might work well for you. Let me know!
> 
> The idea of this strap is for the front to pass by the head tube and then get secured (lashed) to the head tube with velcro, rubber bands, rope, tape, etc.
> 
> This might make a more stable mount for installations that have a long head tube to SS2K distance.
> 
> Anything for dimensions on it is easily adjustable. The front part might need to be a bit beefier yet but I also wanted to try to keep it thin so it attaches easy. If you need anything changed before you print it, let me know.
> 
> https://github.com/doudar/SmartSpin2k/blob/7ebe5c73ab44a7f71848477b43dc71314ea65ccc/Hardware/Mounting_Strap_Rigid.STL

I printed the mounting strap that @jkbaker4 designed and I printed this rigid one.  I think I like the simpler rigid one.  I just put an elastic around the head tube and it was very sturdy.  I may try and design a little clip that replaces the elastic.  Only question:  The part where it meets the bottom case seems to be not as wide as the case, so the short legs can't really be inserted because they are under tension as you pull them apart to go around the case.  Is that intentional to increase the tension and keep it a little stiffer or could I widen them to match the case and make it a little easier to insert?

**Reply by doudar** [reply]
> > I've just uploaded a new style mounting design that might work well for you. Let me know!
> > The idea of this strap is for the front to pass by the head tube and then get secured (lashed) to the head tube with velcro, rubber bands, rope, tape, etc.
> > This might make a more stable mount for installations that have a long head tube to SS2K distance.
> > Anything for dimensions on it is easily adjustable. The front part might need to be a bit beefier yet but I also wanted to try to keep it thin so it attaches easy. If you need anything changed before you print it, let me know.
> > https://github.com/doudar/SmartSpin2k/blob/7ebe5c73ab44a7f71848477b43dc71314ea65ccc/Hardware/Mounting_Strap_Rigid.STL
> 
> I printed the mounting strap that @jkbaker4 designed and I printed this rigid one. I think I like the simpler rigid one. I just put an elastic around the head tube and it was very sturdy. I may try and design a little clip that replaces the elastic. Only question: The part where it meets the bottom case seems to be not as wide as the case, so the short legs can't really be inserted because they are under tension as you pull them apart to go around the case. Is that intentional to increase the tension and keep it a little stiffer or could I widen them to match the case and make it a little easier to insert?

Cool! 

If it's too narrow, that's a design mistake. I'll double check the measurements and re-upload. Glad to hear it works!

How's the Echelon insert working for you? Other than @jkbaker4 baker4 I haven't really had any feedback on that one.

**Reply by doudar** [reply]
Here's a fixed version, 65mm width:
[Mounting_Strap_Rigid.zip](https://github.com/doudar/SmartSpin2k/files/6329878/Mounting_Strap_Rigid.zip)

**Reply by marklogan17** [reply]
The Echelon insert is pretty amazing - especially since you did it from measurements.  I originally put it in the knob cup and tried to push down and I thought it wasn't quite right.  When I looked closer I saw how good it was.  Seriously, it fits perfectly - this is my initial test setup (I'll be reprinting the case at least :) )

![20210417_143243](https://user-images.githubusercontent.com/75054215/115124371-e0778100-9f8f-11eb-8433-f3d4cc9c313c.jpg)

**Reply by marklogan17** [reply]
> Here's a fixed version, 65mm width:
> [Mounting_Strap_Rigid.zip](https://github.com/doudar/SmartSpin2k/files/6329878/Mounting_Strap_Rigid.zip)

Thanks.  Will print today and give it a try.

**Reply by doudar** [reply]
> The Echelon insert is pretty amazing - especially since you did it from measurements. I originally put it in the knob cup and tried to push down and I thought it wasn't quite right. When I looked closer I saw how good it was. Seriously, it fits perfectly - this is my initial test setup (I'll be reprinting the case at least :) )
> 

lol, yah, the purple is a little loud, but what the heck, it's one of a kind! 

That mounting strap looks pretty good! It should be a bunch better with the proper width. That's what happens when you start making designs using copies of copies :) Thanks for the feedback so I could get it fixed. 

As far as I know, you're the first person to print it. The Idea for the simple rigid mount came out of a brainstorming session between @fireyeti @kadaan .

**Reply by kadaan** [reply]
Nice!  Glad that mounting strap looks like it will work!!!

**Reply by marklogan17** [reply]
> lol, yah, the purple is a little loud, but what the heck, it's one of a kind!
> 
> That mounting strap looks pretty good! It should be a bunch better with the proper width. That's what happens when you start making designs using copies of copies :) Thanks for the feedback so I could get it fixed.
> 
> As far as I know, you're the first person to print it. The Idea for the simple rigid mount came out of a brainstorming session between @fireyeti @kadaan .

Cheap PLA is cheap PLA - sometimes colour is less important than price.  LOL.  I printed the top cover in clear so that I can hopefully at least see the lights through it.  The purple will, umm, not be on the final product unless I get really lazy...

I'm thinking a little latch that goes around the head tube and just basically snaps over the rigid mounting strap would be nice replacement for the elastic band eventually, really just for aesthetics, not for function.  However I'll worry about that once I have everything working or I'm bored one evening...

**Reply by marklogan17** [reply]
> Here's a fixed version, 65mm width:
> [Mounting_Strap_Rigid.zip](https://github.com/doudar/SmartSpin2k/files/6329878/Mounting_Strap_Rigid.zip)

This strap worked great.  I think the whole setup for the Echelon is pretty solid at this point.  Now just be patient for electronic pieces :)

**Reply by doudar** [reply]
> > Here's a fixed version, 65mm width:
> > [Mounting_Strap_Rigid.zip](https://github.com/doudar/SmartSpin2k/files/6329878/Mounting_Strap_Rigid.zip)
> 
> This strap worked great. I think the whole setup for the Echelon is pretty solid at this point. Now just be patient for electronic pieces :)

Nice!

I'm sure you'll have some questions, when you get to that point, feel free to start a new discussion (this one's getting a bit long). 

Can't wait! Thanks Mark!

**Reply by marklogan17** [reply]
Last one on this thread :) I did a quick design of a clasp to replace the elastic band.  Just goes around the head tube and pops over the mounting strap.  Very simple and can probably be improved, but I think it holds the strap with just enough tension and makes it a bit more 'complete' than the elastic band...

![20210419_082951](https://user-images.githubusercontent.com/75054215/115239236-8213e500-a0ec-11eb-9341-8056968d56fc.jpg)
[EchelonStrapClasp.stl.zip](https://github.com/doudar/SmartSpin2k/files/6336161/EchelonStrapClasp.stl.zip)

**Reply by doudar** [reply]
I'm starting to like that rigid mount more and more! Are you able to provide whatever cad file you have for the retainer clip? I'll add it to the repository (unless you can do a PR, which is better).  

To do a PR, fork this repository ( which creates a copy in your github account), then upload the files to the appropriate hardware section in your repository, edit the changelod.md (like "designed new retainer clip"), then select "create pull request" with this repository as the target. If you have questions, I'll walk you through it.

**Reply by marklogan17** [reply]
> I'm starting to like that rigid mount more and more! Are you able to provide whatever cad file you have for the retainer clip? I'll add it to the repository (unless you can do a PR, which is better).
> 
> To do a PR, fork this repository ( which creates a copy in your github account), then upload the files to the appropriate hardware section in your repository, edit the changelod.md (like "designed new retainer clip"), then select "create pull request" with this repository as the target. If you have questions, I'll walk you through it.

I just did it in Tinkercad so the output options are a little limited.  The STL is in the zip on the previous post.  I can export it to OBJ or GLB if either of them are better.  Go ahead and add to the repository - rename it to be something generic if you think it will fit more bikes.  I can give you the dimensions if its just easier to re-create in a more powerful modelling tool.

**Reply by doudar** [reply]
Could you do just a quick 2d sketch of it with dimensions? I'll see if I can then draw it up. Probably 2 views. One top down, and another from the end so I can see the height and the pass through hole dimensions.  Doesn't need to be anything fancy.

**Reply by marklogan17** [reply]
See attached.  
Other sizes:
Back bar: 3mm thick, 10mm high (the 13mm in the picture is actually the latch) and 80mm long
Arms: 3mm thick, 38mm long, 7mm wide
Vertical latches: 2mm thick, 7mm wide, 13mm high
'latch': just a 2mmx2mm triangle - so that effectively leaves 8mm of room for the 7mm wide 'rigid strap'

I probably didn't describe this well so let me know what other info you need :)


![Screen Shot 2021-04-19 at 9 29 07 AM](https://user-images.githubusercontent.com/75054215/115245329-d15d1400-a0f2-11eb-81bf-d8c7faf52ef2.png)
![Screen Shot 2021-04-19 at 9 29 33 AM](https://user-images.githubusercontent.com/75054215/115245335-d28e4100-a0f2-11eb-82b9-3819a2ca04c1.png)

**Reply by doudar** [reply]
Awesome thx!

**Reply by doudar** [reply]
What printing method your used for the overhangs on the hooks. Supports?

**Reply by marklogan17** [reply]
> What printing method your used for the overhangs on the hooks. Supports?

Honestly for this quick test and that small overhang I didn't even bother with supports, just let it go. It was a touch ugly but actually worked fine.  I thought about reversing the triangle so that supports wouldn't be necessary and you would get a clean print - it would probably be the best way and I'm sure it would hold just as well.

**Reply by doudar** [reply]
I added yours to develop as well as another completely captive one. If you'd like to burn more filament and time, you can test mine too :)

**Reply by marklogan17** [reply]
I gave it a quick try - the arms are a little short to go around my head tube and over the rigid strap.  The space between the back and the front on yours is just over 30mm and it needs to be 38mm to go over and latch.  If you were intending the bar to slide into those rectangles on the front then the rectangles are slightly off.  Height wise I think they're OK, but width wise the rigid strap is just over 7mm, let's call it 7.5mm and the width of the rectangles on yours are approx 6mm.

**Reply by marklogan17** [reply]
@doudar I think I need to apologize :)  I just tried your updated retention clip and I think I unintentionally gave you some extra work...   I wasn't sure if you were intending to latch over top like my original, or to slide the rigid strap through.  So my comment made it look like you needed two changes when in fact you only needed the one...  The changes you made to the squares on the front are perfect and now the rigid strap slides through perfectly.  However the arms are now a little long and its not 'tight' on the head tube.  Your original 30mm length would be perfect.  I measured and the head tube is about 31.5mm so 30mm is good because it will flex around the head tube a little and really keep it in place.  Picture of new version below so you can see the gap I'm talking about. (also note the black bottom case now - no more purple, now black ABS :) )

![20210424_132749](https://user-images.githubusercontent.com/75054215/115969161-951e1f00-a509-11eb-99e1-41571617ef7d.jpg)

**Reply by doudar** [reply]
> @doudar I think I need to apologize :)  I just tried your updated retention clip and I think I unintentionally gave you some extra work...   I wasn't sure if you were intending to latch over top like my original, or to slide the rigid strap through.  So my comment made it look like you needed two changes when in fact you only needed the one...  The changes you made to the squares on the front are perfect and now the rigid strap slides through perfectly.  However the arms are now a little long and its not 'tight' on the head tube.  Your original 30mm length would be perfect.  I measured and the head tube is about 31.5mm so 30mm is good because it will flex around the head tube a little and really keep it in place.  Picture of new version below so you can see the gap I'm talking about. (also note the black bottom case now - no more purple, now black ABS :) )
> 
> 
> 
> ![20210424_132749](https://user-images.githubusercontent.com/75054215/115969161-951e1f00-a509-11eb-99e1-41571617ef7d.jpg)
> 
> 

Hehe, no problem. It's an easy change.

The new case is looking great!

**Reply by marklogan17** [reply]
@jkbaker4 what settings did you use within SmartSpin for steps and multiplier?  Trying to find what will work for a single button press moving to a new resistance level.  Trouble is that I'm not sure that its a constant amount of turning needed on the bike - i.e. it may take more turning the higher resistance we go.

**Reply by jkbaker4** [reply]
Good Morning, the rate is constant from 0 to 30.  Its basically about 3/4 turn to go from 15 to 16.

**Reply by kadaan** [reply]
@marklogan17 You should be able to easily test this with the SS2K website and the shifter buttons.

**Reply by marklogan17** [reply]
> @marklogan17 You should be able to easily test this with the SS2K website and the shifter buttons.

Yes I'm doing that.  So far it looks like maybe 650-700 steps will do it.  I was just curious what @jkbaker4 had found since he had the same model of bike.

**Reply by jkbaker4** [reply]
My shift step is 750 and seems to be working.  

![SmartSpin2k Settings](https://user-images.githubusercontent.com/40582757/118057741-8386a500-b35a-11eb-9d22-fa9645b5bc6d.JPG)
